csharp/src/Ice/WSEndpoint.cs

```diff
@@ -148,11 +148,25 @@ internal WSEndpoint(InputStream istr, Communicator communicator, Transport trans
             string endpointString)
             : base(communicator, transport, protocol, host, port, options, oaEndpoint, endpointString)
         {
-            if (options.TryGetValue("resource", out string? value))
+            string? value = null;
+            if (options.TryGetValue("resource", out value))
             {
+                // The resource value (as supplied in a URI string) must be percent-escaped with '/' separators
+                // We keep it as-is, and will marshal it as-is.
                 _resource = value;
                 options.Remove("resource");
             }
+            else if (options.TryGetValue("option", out value))
```

Comment:
It's not clear to me how the universal "option" option works. Is it supposed to contain a sequence of options which are supposed to be specified with an ordered comma separated string? Why can't I simply specify the option dictionary instead? Shouldn't the following two URIs be equivalent? 

```
ice+ws://localhost:10002/hello?resource=/foo
ice+universal://localhost:10002/hello?transport=4&resource=/foo
```

Reply:
ice+universal is used to represent endpoints for transports that are not supported by the current Ice communicator. It also supports the transports that are supported by the current Ice communicator, but it's definitely not its main purpose. And in this "supported transport" case, when you parse an ice+universal URI, you get a tcp, ws (...) endpoint, not a universal aka opaque endpoint.

The options for the ice+universal scheme are closely related to the Ice 2.0 encoding of endpoints. ice+universal needs the transport number (short), the host (string), port number (ushort) and a sequence<string> (options). ice+universal's query option named "option" corresponds to this sequence<string>--the marshaled option strings. The other ice+universal's query option is "transport" and is mandatory. ice+universal does not have any local-only query option. Selecting a supported and/or known transport does not extend the query options of ice+universal; for example "resource" is a not an ice+universal query option, even if you select transport=ws.